src/qibo/abstractions/abstract_gates.py

```diff
@@ -205,6 +207,10 @@ def on_qubits(self, *q) -> "Gate":
             A :class:`qibo.abstractions.gates.Gate` object of the original gate
             type targeting the given qubits.
         """
+        # Note that q is interpreted as a map from the original qubits to the new ones, e.g.
+        # gates.CNOT(2, 3)._on_qubits(0, 1, 2, 3) is equivalent to gates.CNOT(2, 3)
+        # gates.CNOT(2, 3)._on_qubits(1, 2, 3, 0) is equivalent to gates.CNOT(3, 0)
+        # It is required for `len(q)` to be greater than the max qubit id of the original gate
```

Comment:
I believe this comment can be converted to an actual check.

Reply:
Ok, I can do it, but it would require something like ``if len(q) <= max(self.qubits): raise_error(...)`` to be executed for every call.
Is the overhead worth it? Because if it is called only by the ``AbstractCircuit.on_qubits()`` method, a more strict check is already performed once before calling ``Gate._on_qubits()`` for each gate in the circuit:
https://github.com/qiboteam/qibo/blob/9e0002072ec43d1d438c2a19733acadf122357b9/src/qibo/abstractions/circuit.py#L176-L181

Reply:
Ok, that is a good point indeed. If the bad configurations are caught by the `AbstractCircuit.on_qubits` this hidden/private method can stay as it is.